### PR TITLE
fix: send numbers or bools as payloads

### LINKED ISSUE #1461: Accept Boolean as body in send()
- [ x ] Used appropriate template for the issue type
- [ x ] Searched both open and closed issues for duplicates of this issue
- [ x ] Title adequately and _concisely_ reflects the feature or the bug

# Bug Report

## Restify Version

5.x

## Node.js Version

Any

## Expected behaviour

Under for 4.x, you can respond with `res.json(200, true)`. This sets body to `true`. This is no longer possible under 5.x

Related: https://github.com/restify/node-restify/issues/1406

## Actual behaviour

Body has no content.

## Repro case

`res.json(200, true)`

## Cause

`./lib/response.js` on line 272 checks for the type and filters out anything that isn't an object or a string.

## Are you willing and able to fix this?

Already did. Ready to make a PR once this is given the green light.

https://github.com/revnode/node-restify/commit/8c598eeb8a5315f5758f330791991eb198b1e53b

### PR DESCRIPTION
**Pull request by DonutEspresso**
Fixes #1461 and #1607

### CODE REVIEW

**Comment by DonutEspresso** [comment]
@retrohacker is master still our target for 6.x? I see the new middleware/routing system is targeted to land on next.

**Comment by DonutEspresso** [comment]
Sorry @hekike I had to remove your fancy variadic parsing - doesn't seem possible to disambiguate a bool payload from the format arg!

File: tools/mk/Makefile.targ [hunk 1]
```diff
@@ -122,6 +122,10 @@ check-bash: $(BASH_FILES:%=%.bashchk) $(BASH_FILES:%=%.bashstyle)
 %.bashstyle: %
 	$(BASHSTYLE) $^
 
+.PHONY: check-eslint-fix
```
Comment (hekike) [inline tools/mk/Makefile.targ]: we have a `make fix-style` that reprints everything with `prettier`, probably more accurate than ESLint fix: https://github.com/restify/node-restify/blob/master/tools/mk/Makefile.targ#L137

Comment (DonutEspresso) [reply tools/mk/Makefile.targ]: Oh nice! I completely forgot you added that! I'll remove this target  👍

**Pull request merged**